src/dappsExplorer/search/DAppsExplorerScreenSearch.tsx

```diff
@@ -0,0 +1,327 @@
+import useDappFavoritedToast from 'src/dappsExplorer/useDappFavoritedToast'
+import useDappInfoBottomSheet from 'src/dappsExplorer/useDappInfoBottomSheet'
+import useOpenDapp from 'src/dappsExplorer/useOpenDapp'
+import { calculateSearchScore } from 'src/dappsExplorer/utils'
+import { currentLanguageSelector } from 'src/i18n/selectors'
+import Help from 'src/icons/Help'
+import DrawerTopBar from 'src/navigator/DrawerTopBar'
+import { TopBarIconButton } from 'src/navigator/TopBarButton'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+const AnimatedSectionList =
+  Animated.createAnimatedComponent<SectionListProps<DappV2, SectionData>>(SectionList)
+
+interface SectionData {
+  data: (DappV1WithCategoryName | DappV2WithCategoryNames)[]
+  category: string
+}
+
+export function DAppsExplorerScreenSearch() {
+  const { t } = useTranslation()
+  const insets = useSafeAreaInsets()
+
+  const sectionListRef = useRef<SectionList>(null)
+  const scrollPosition = useRef(new Animated.Value(0)).current
+
+  const onScroll = Animated.event([{ nativeEvent: { contentOffset: { y: scrollPosition } } }])
+  const dispatch = useDispatch()
+  const loading = useSelector(dappsListLoadingSelector)
+  const error = useSelector(dappsListErrorSelector)
+  const categories = useSelector(dappsCategoriesAlphabeticalSelector)
+  const dappFavoritesEnabled = useSelector(dappFavoritesEnabledSelector)
+  const dappsMinimalDisclaimerEnabled = useSelector(dappsMinimalDisclaimerEnabledSelector)
+  const dappList = useSelector(dappsListSelector)
+  const dappListWithCategoryNames = useSelector(dappListWithCategoryNamesSelector)
+  const language = useSelector(currentLanguageSelector)
+  const favoriteDappsById = useSelector(favoriteDappIdsSelector)
+
+  // Some state lifted up from all and favorite sections
+  const [searchTerm, setSearchTerm] = React.useState('')
+  const [favoriteResultsEmpty, setFavoriteResultsEmpty] = React.useState(false)
+  const [allResultEmpty, setAllResultEmpty] = React.useState(false)
+
+  const { onSelectDapp, ConfirmOpenDappBottomSheet } = useOpenDapp()
+  const { onFavoriteDapp, DappFavoritedToast } = useDappFavoritedToast(sectionListRef)
+  const { openSheet, DappInfoBottomSheet } = useDappInfoBottomSheet()
+
+  useEffect(() => {
+    dispatch(fetchDappsList())
+    ValoraAnalytics.track(DappExplorerEvents.dapp_screen_open)
+  }, [])
+
+  const allSectionResults: SectionData[] = React.useMemo(() => {
+    const allResultsParsed = parseResultsIntoAll(
+      dappListWithCategoryNames,
+      searchTerm,
+      favoriteDappsById
+    )
+    if (allResultsParsed.length === 0) {
+      setAllResultEmpty(true)
+    } else {
+      setAllResultEmpty(false)
+    }
+    return allResultsParsed
+  }, [dappList, searchTerm, favoriteDappsById])
+
+  const emptyListComponent = React.useMemo(() => {
+    if (allResultEmpty && favoriteResultsEmpty) return null
```

Comment:
Shouldn't we display something?

Reply:
This is the weird case where favorites section is empty and the all section is empty. The favorites will display a no results message and we want to display nothing here.